chain/client/tests/process_blocks.rs

```diff
@@ -2879,3 +2878,107 @@ mod access_key_nonce_range_tests {
         ));
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "protocol_feature_fix_storage_usage")]
+mod storage_usage_fix_tests {
+    use super::*;
+    use borsh::BorshDeserialize;
+    use near_primitives::types::AccountId;
+    use near_primitives::version::ProtocolFeature;
+    use near_store::TrieUpdate;
+    use std::rc::Rc;
+
+    fn process_blocks_with_storage_usage_fix(
+        chain_id: String,
+        check_storage_usage: fn(AccountId, u64, u64),
+    ) {
+        let epoch_length = 5;
+        let mut genesis = Genesis::test(vec!["test0", "near"], 1);
+        genesis.config.chain_id = chain_id;
+        genesis.config.epoch_length = epoch_length;
+        genesis.config.protocol_version = ProtocolFeature::FixStorageUsage.protocol_version() - 1;
+        let genesis_height = genesis.config.genesis_height;
+        let chain_genesis = ChainGenesis::from(&genesis);
+        let mut env =
+            TestEnv::new_with_runtime(chain_genesis, 1, 1, create_nightshade_runtimes(&genesis, 1));
+        for i in 1..=16 {
+            let (encoded_chunk, merkle_paths, receipts) =
+                create_chunk_on_height(&mut env.clients[0], i);
+
+            let mut chain_store =
+                ChainStore::new(env.clients[0].chain.store().owned_store(), genesis_height);
+            env.clients[0]
+                .shards_mgr
+                .distribute_encoded_chunk(
+                    encoded_chunk.clone(),
+                    merkle_paths.clone(),
+                    receipts.clone(),
+                    &mut chain_store,
+                )
+                .unwrap();
+
+            let mut block = env.clients[0].produce_block(i).unwrap().unwrap();
+
+            let validator_signer = InMemoryValidatorSigner::from_seed(
+                &"test0".to_string(),
+                KeyType::ED25519,
+                &"test0".to_string(),
+            );
+
+            block.mut_header().get_mut().inner_rest.latest_protocol_version =
+                ProtocolFeature::FixStorageUsage.protocol_version();
```

Comment:
Could you add a comment on the difference between this and `TestEnv::produce_block`?